### ISSUE TITLE
Auto update on Windows not work

### ISSUE THREAD

**Issue by Stephen988** [tailscale/tailscale#12559]
### What is the issue?

Hello,
on Windows client enabling "Automatically install updates" not work, don't appear any alert or message, for notify new version and I need every time to update manually

### OS

Windows

### OS version

Windows 10 and 11

### Tailscale version

any version

**Comment by agottardo** [comment]
Can you attach a bug report identifier?

**Comment by Stephen988** [comment]
> Can you attach a bug report identifier?

at the moment no, but it's very simple, install the penultimate version and enable auto update.
then click on the tray icon and it show there is a new version available, but it will never update automatically.

**Comment by agottardo** [comment]
We ask for a bug report ID because it would allow us to look at your client logs, in order to determine why updates are not being fetched.

**Comment by Stephen988** [comment]
how to add bug report?
![image](https://github.com/tailscale/tailscale/assets/64849759/d3d6f503-d7a7-49ed-914b-54b2f5eddad6)

**Comment by agottardo** [comment]
https://tailscale.com/kb/1227/bug-report

**Comment by Stephen988** [comment]
> https://tailscale.com/kb/1227/bug-report
> 
> > 1. Hold the Ctrl key and right-click the Tailscale icon in the menu bar.
> > 2. Click Debug and then click Bug report.
> > 3. Click Copy report ID to copy the bug report.

Hello,
here the bug report:

Bugreport marker created:

	BUG-de515c3133afd549e56d9ce622a5e1a4c6a0445752163c267c6e5a075f83b825-20240706092008Z-874518f2f4d45575

**Comment by Stephen988** [comment]
Hi @agottardo my last message is ok for investigate about failing automatic update?

**Comment by jksmurf** [comment]
I have the same issue. I have auto updates enabled in the console and install updates automatically on the taskbar icon.

Bugreport marker created:

	BUG-d6929a9c94da8849ebf8ce27ef73be9cbce89886728f480e07695f210494e588-20241011223704Z-82ef4f806d20ad49

Email that to support@tailscale.com with a description of the problem.

![Enabled](https://github.com/user-attachments/assets/eb6065cd-486b-4c1f-862b-429dc5f2413c)
![AU](https://github.com/user-attachments/assets/87b856ef-3d90-4967-80c3-9e705b99d184)

**Comment by jksmurf** [comment]
Possible solution?

So the solution [might be here](https://www.reddit.com/r/Tailscale/comments/1c5mxw9/comment/kzwhwaj/) (I am not sure so if someone could confirm?):

1. Check current status by running a CMD Window in Windows and typing in "_tailscale debug prefs_", enter.
2. If in the list you see:

```
"AutoUpdate": {

"Check": false,

"Apply": true

},
```
if the "Check" part is false it may not be checking, even though apply is true?

3. To change Check to true, in a CMD Window in Windows, type in "_tailscale set --update-check=true_", enter.

HTH

**Comment by EDIflyer** [comment]
Same issue here - autoupdate enabled but none of my Windows machines updated to 1.78.1 - BUG-9068247bcce9ac11cbce71d4cc11cdae03dfe30e9ad9768f586c2aa4a1ebb0c6-20241209222509Z-2e1f9824aa1cea37

I also can't seem to apply the update as it just hangs at the logging when doing a `tailscale update` from an admin command prompt and when I tried from the downloaded msi it initially said other operation in progress. Then tried a repair - eventually completed but I'm still on 1.76.6.

![image](https://github.com/user-attachments/assets/439a0b4d-9447-4f36-8dbe-6625bf80b7c5)

**Comment by jksmurf** [comment]
Mine still doesn’t auto-update.

I even completely uninstalled them reinstalled it, but same issue. Shows the latest, but still doesn’t auto-update it.

**Comment by Stephen988** [comment]
> Mine still doesn’t auto-update.
> 
> I even completely uninstalled them reinstalled it, but same issue. Shows the latest, but still doesn’t auto-update it.
> 
> I can update it manually though.

yes this bug continue to not be solved ☹️
I have scheduled a script to run monthly and update via command prompt

**Comment by EDIflyer** [comment]
Just to check if auto-update is working OK for folks on Windows now? I've not yet noticed any machine update automatically - I've either had to install it manually via `tailscale update` or initiate it from the management console. The box is definitely ticked.

**Comment by Stephen988** [comment]
> Just to check if auto-update is working OK for folks on Windows now? I've not yet noticed any machine update automatically - I've either had to install it manually via `tailscale update` or initiate it from the management console. The box is definitely ticked.

no seems this bug will never patched.

**Comment by jksmurf** [comment]
> Just to check if auto-update is working OK for folks on Windows now? I've not yet noticed any machine update automatically - I've either had to install it manually via `tailscale update` or initiate it from the management console. The box is definitely ticked.

Mine doesn’t auto/update either. Even if I leave it for a white. Even after I tried a complete uninstall reinstall.

**Comment by EDIflyer** [comment]
Any update on auto-update not working on Windows?

![Image](https://github.com/user-attachments/assets/ae19dab0-6a69-4c62-8415-a982e95430fb)

**Comment by Stephen988** [comment]
> Any update on auto-update not working on Windows?
> 
> ![Image](https://github.com/user-attachments/assets/ae19dab0-6a69-4c62-8415-a982e95430fb)

give it up, there is no hope.

**Comment by nickkhyl** [comment]
As mentioned before, we usually don't enable auto-updates for a new release on its release day. Clients who've opted in for auto updates typically begin installing it about a week later. Until then, the client sees the new release and may notify the user, but it won’t install it automatically. In some cases, we might choose not to enable auto-updates for a given release and opt to do so with a subsequent point release. We also try to defer updates if the device is busy (for example, if there's heavy traffic or an active SSH session). This behavior is intentional; for users who prefer a stable release, it makes little difference whether they receive the new version today or in a week.

If I understand your request correctly, you want the new client installed as soon as the new version is released and available for download. While I can see why some folks might want that, it's not a priority for us right now. That said, there are a couple of options:
- Switch to the [unstable track](https://pkgs.tailscale.com/unstable/#windows): with unstable, you'll receive updates as soon as they're available. You're more likely to encounter bugs, but that might be acceptable if you value early access to new features and improvements.
- Automate it on your end (e.g., using Task Scheduler). If you encounter any issues with installing updates in general, let's file and track those separately.

---

@EDIflyer: based on the logs, it appears you've installed v1.80 on all 3 active Windows devices before we enabled auto-updates to v1.80.2 on 2025-02-19. Similarly, you installed v1.82 on 2 out of 3 Windows devices before we enabled auto-updates on 2025-04-01. The third device was offline for over a week and was updated manually about 14 minutes after coming online, so it didn’t have a chance to install updates automatically.

---

@Stephen988: based on the logs, you have auto-updates disabled on both your Windows devices.

**Comment by EDIflyer** [comment]
@nickkhyl thanks - I didn't realise there was such a long delay between release and auto-updates being enabled.  I can understand the reasoning but was confused when I would see the release announcement (I follow the Tailscale blog 😉) and then days later my clients still hadn't updated.  It might be worth mentioning somewhere (either in the blog or the UI) that auto-updates may take a week or two to happen.

**Comment by pjdubya** [comment]
What is the logic that it's using today? I have a device that I don't power up very often, but did yesterday after being off for about 4 months. It's running 1.84.2 now and reports that the latest version available is 1.90.6, and although it was left running overnight and auto-update was activated long ago, it hasn't updated. I don't see a way to even trigger it to update manually in the UI. I can understand that you wouldn't want to upgrade everyone to the latest right away necessarily, I would think it would always at least seek to upgrade months-out-of-date installations asap to whatever the latest release that no longer has auto update timing restrictions in place. In this case, 1.90.6 has been out for 2 weeks so I wouldn't think there are even any timing restrictions in place on that one anyway. Just curious what the real logic is here so I can know if I should just get used to manually triggering update via cmd for this device.

**Comment by jksmurf** [comment]
> What is the logic that it's using today? … Just curious what the real logic is here so I can know if I should just get used to manually triggering update via cmd for this device.

I agree, it just seems this auto update logic is somehow flawed. My post above was Oct 24 and my windows PCs still do not update after a reasonable period of time, like many here.
